File: src/dataset/cpp/beer-song_exercism-tqa236/claude/ClaudeSonnet4_beer_song_v2.cpp
#include "beer_song.h"

std::string beer_song::verse(int verse_num) {
  switch (verse_num) {
    case 0:
      return "No more bottles of beer on the wall, no more bottles of beer.\nGo "
             "to the store and buy some more, 99 bottles of beer on the wall.\n";
    case 1:
      return "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n";
    case 2:
      return "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n";
    default: {
      const std::string num_str = std::to_string(verse_num);
      const std::string prev_num_str = std::to_string(verse_num - 1);
      return num_str + " bottles of beer on the wall, " + num_str +
             " bottles of beer.\nTake one down and pass it around, " +
             prev_num_str + " bottles of beer on the wall.\n";
    }
  }
}

std::string beer_song::sing(int start, int end) {
  std::string lyrics;
  lyrics.reserve((start - end + 1) * 150); // Pre-allocate approximate memory
  
  for (int i = start; i >= end; --i) {
    lyrics += beer_song::verse(i);
    if (i > end) {
      lyrics += '\n';
    }
  }
  return lyrics;
}

std::string beer_song::sing(int start) { 
  return beer_song::sing(start, 0); 
}

File: src/dataset/javascript/beer-song_exercism-elenamountz/claude/ClaudeSonnet4_beer-song_v2.js
export default class Beer{
  constructor(){  }

  static verse(n){
    if(n === 0) return `No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n`;
    if(n === 1) return `1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n`;
    if(n === 2) return `2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n`;
    return `${n} bottles of beer on the wall, ${n} bottles of beer.\nTake one down and pass it around, ${n-1} bottles of beer on the wall.\n`;
  }

  static sing(from = 99, to = 0){
    const verses = [];
    for(let i = from; i >= to; i--){
      verses.push(this.verse(i));
    }
    return verses.join('\n').slice(0, -1);
  }
}